src/explore-education-statistics-common/src/modules/table-tool/components/TableToolWizard.tsx

```diff
@@ -0,0 +1,359 @@
+import PublicationSubjectForm, {
+  PublicationSubjectFormSubmitHandler,
+} from '@common/modules/table-tool/components/PublicationSubjectForm';
+import TimePeriodForm, {
+  TimePeriodFormSubmitHandler,
+} from '@common/modules/table-tool/components/TimePeriodForm';
+import Wizard from '@common/modules/table-tool/components/Wizard';
+import WizardStep from '@common/modules/table-tool/components/WizardStep';
+import { Dictionary } from '@common/types/util';
+import mapValues from 'lodash/mapValues';
+import React, { ReactElement } from 'react';
+import { TableHeadersConfig } from '@common/modules/full-table/utils/tableHeaders';
+import { TableHeadersFormValues } from '@common/modules/table-tool/components/TableHeadersForm';
+import {
+  DateRangeState,
+  getDefaultSubjectMeta,
+  initialiseFromInitialQuery,
+  mapLocations,
+  tableGeneration,
+} from './utils/tableToolHelpers';
+
+interface Publication {
+  id: string;
+  title: string;
+  slug: string;
+}
+
+interface TableToolState {
+  query: TableDataQuery | undefined;
+  createdTable: FullTable | undefined;
+  validInitialQuery: TableDataQuery | undefined;
+  dateRange: DateRangeState;
+  locations: Dictionary<LocationFilter[]>;
+  subjectId: string;
+  subjectMeta: PublicationSubjectMeta;
+}
+
+type PartialState = {
+  [P in keyof TableToolState]?: TableToolState[P];
+};
+
+export interface FinalStepProps {
+  publication?: Publication;
+  createdTable?: FullTable;
+  tableHeadersConfig?: TableHeadersConfig;
+  tableHeaders?: TableHeadersFormValues;
+  query?: TableDataQuery;
+}
+
+interface Props {
+  themeMeta: ThemeMeta[];
+  publicationId?: string;
+  releaseId?: string;
+
+  finalStep?: (props: FinalStepProps) => ReactElement;
+
+  initialQuery?: TableDataQuery;
+  onTableConfigurationChange?: (props: FinalStepProps) => void;
+}
+
+const TableToolWizard = (props: Props) => {
+  const {
+    themeMeta,
+    publicationId,
+    releaseId,
+    finalStep,
+    initialQuery,
+    onTableConfigurationChange,
+  } = props;
+
+  const [publication, setPublication] = React.useState<Publication>();
+  const [subjects, setSubjects] = React.useState<PublicationSubject[]>([]);
+
+  const [initialStep, setInitialStep] = React.useState(1);
+
+  const getInitialState = () => {
+    return {
+      initialStep: 1,
+      subjectId: '',
+      subjectMeta: getDefaultSubjectMeta(),
+      locations: {},
+      dateRange: {},
+      createdTable: undefined,
+      query: undefined,
+      validInitialQuery: undefined,
+    };
+  };
+
+  const [tableToolState, setTableToolState] = React.useState<TableToolState>(
+    () => {
+      return getInitialState();
+    },
+  );
+
+  React.useEffect(() => {
+    if (releaseId) {
+      tableBuilderService
+        .getReleaseMeta(releaseId)
+        .then(({ subjects: releaseSubjects }) => {
+          setSubjects(releaseSubjects);
+        });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why do we need this?  Which dependencies is it complaining that need representing here?

Reply:
It's not any more